src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,128 @@
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.Test;
+
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Feel free to remove this class and create your own.
+ */
+public class RegistrationServiceTest {
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+    private User user = new User();
+
+    @Test
+    void registerNullUser_NotOk() {
+        RegistrationException exception = assertThrows(RegistrationException.class, () -> {
+            registrationService.register(null);
+        });
+
+        assertEquals("User is null.", exception.getMessage());
+    }
+
+    @Test
+    void registerLoginNull_NotOk() {
+        user.setLogin(null);
+        user.setPassword("validPassword");
+        user.setAge(20);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+        assertEquals("Login can't be null.", exception.getMessage());
+    }
+
+    @Test
+    void registerPasswordNull_NotOk() {
+        user.setLogin("validLogin");
+        user.setPassword(null);
+        user.setAge(20);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+        assertEquals("Password can't be null.", exception.getMessage());
+    }
+
+    @Test
+    void registerUserAgeNull_NotOk() {
+        user.setLogin("validLogin");
+        user.setPassword("validPassword");
+        user.setAge(null);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+        assertEquals("Age can't be null.", exception.getMessage());
+    }
+
+    @Test
+    void registerLogin_NotOk() {
+        String[] invalidLogins = {"1", "ab", "son", "hell", "user"};
+
+        for (String login : invalidLogins) {
+            user.setLogin(login);
+            user.setPassword("validPassword");
+            user.setAge(20);
+
+            RegistrationException exception = assertThrows(RegistrationException.class, () -> {
+                registrationService.register(user);
+            });
+
+            assertEquals("Login length must be at least 6 characters.", exception.getMessage());
+        }
+    }
+
+    @Test
+    void registerPassword_NotOk() {
+        String[] invalidPasswords = {"hello", "hell", "user"};
+
+        for (String password : invalidPasswords) {
+            user.setLogin("validLogin");
+            user.setPassword(password);
+            user.setAge(20);
+
+            RegistrationException exception = assertThrows(RegistrationException.class, () -> {
+                registrationService.register(user);
+            });
+
+            assertEquals("Password length must be at least 6 characters.", exception.getMessage());
```

Comment:
The test `registerPassword_NotOk` correctly checks for the password length edge cases, but it would be better to include a test for a password with the exact minimum length (6 characters) to verify that it is considered valid.